tests/conftest.py

```diff
@@ -366,18 +367,39 @@ def shutdown(self):
                                 + " and had to be killed") from None
 
 
+def kill_all_existing_windows(show_warnings=True):
+    xlsclients = subprocess.run(['xlsclients', '-l'], stdout=subprocess.PIPE)
+    assert xlsclients.returncode == 0
+    clients = []
+    for l in xlsclients.stdout.decode().splitlines():
+        m = re.match(r'Window (0x[0-9a-fA-F]*):', l)
+        if m:
+            clients.append(m.group(1))
+    if len(clients) > 0:
```

Comment:
You could reduce the indendation here by saying `if clients and show_warnings:` here, and moving the `for` one level up (it doesn't care whether the list is empty).